src/motion_detector.py

```diff
@@ -58,15 +58,42 @@ def validate_config(cls, config: ServiceConfig) -> Sequence[str]:
         source_cam = config.attributes.fields["cam_name"].string_value
         if source_cam == "":
             raise ValueError("Source camera must be provided as 'cam_name'")
-        min_boxsize = config.attributes.fields["min_box_size"].number_value
-        if min_boxsize < 0:
-            raise ValueError("Minimum bounding box size should be a positive integer")
-        sensitivity = config.attributes.fields["sensitivity"].number_value
-        if sensitivity < 0 or sensitivity > 1:
-            raise ValueError("Sensitivity should be a number between 0 and 1")
-        max_box_size = config.attributes.fields.get("max_box_size")
-        if max_box_size is not None and max_box_size.number_value <= 0:
-            raise ValueError("Maximum bounding box size, if present, must be a positive integer")
+
+        # You must specify either a min_box_size or a min_box_percent.
+        min_box_size    = config.attributes.fields.get("min_box_size")
+        min_box_percent = config.attributes.fields.get("min_box_percent")
+        if min_box_size is None:
+            if min_box_percent is not None:
+                min_box_val = min_box_percent.number_value
+                if min_box_val < 0.0 or min_box_val > 1.0:
+                    raise ValueError("Minimum bounding box percent should be between 0.0 and 1.0")
+        else:  # min_box_size is not None
+            if min_box_percent is not None:
+                raise ValueError("Cannot specify the minimum box in both pixels and percentages")
+            min_box_val = min_box_size.number_value
+            if min_box_val < 0:
+                raise ValueError("Minimum bounding box size should be a non-negative integer")
+
+        sensitivity = config.attributes.fields.get("sensitivity")
+        if sensitivity is not None:
+            sensitivity = sensitivity.number_value
+            if sensitivity < 0 or sensitivity > 1:
+                raise ValueError("Sensitivity should be a number between 0.0 and 1.0")
+
+        max_box_size    = config.attributes.fields.get("max_box_size")
+        max_box_percent = config.attributes.fields.get("max_box_percent")
+        if max_box_size is None:
+            if max_box_percent is not None:
+                max_box_val = max_box_percent.number_value
+                if max_box_val < 0.0 or max_box_val > 1.0:
+                    raise ValueError("Maximum bounding box percent should be between 0.0 and 1.0")
+        else:  # max_box_size is not None
+            if max_box_percent is not None:
+                raise ValueError("Cannot specify the maximum box in both pixels and percentages")
+            max_box_val = max_box_size.number_value
+            if max_box_val < 0:
+                raise ValueError("Maximum bounding box size should be a non-negative integer")
```

Comment:
I don't like how lines 83-95 are basically a copy of lines 63-75. I'm willing to refactor if you want, but I'm trying to take Bijan's advice to heart and not spend too much time cleaning things up yet.

Reply:
After cleaning up the use of the `.fields` non-dict, this has been greatly simplified, and even the linter is happy. No need to refactor further.